From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zyr1xx <nadejdinkostya@gmail.com>
Date: Fri, 31 Jul 2026 00:38:38 +0300
Subject: [PATCH] Hook database/redis service into server lifecycle


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index a19eee14eccaa6cbff42b054d803a41a4f00443f..047431262f0f6874a587b6db8032ee51f1015446 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -161,8 +161,19 @@ import net.minecraft.world.level.storage.WorldData;
 import net.minecraft.world.phys.Vec2;
 import net.minecraft.world.phys.Vec3;
 import org.slf4j.Logger;
+import ru.lewis.point.api.DatabaseService;
+import ru.lewis.point.api.PointAPI;
+import ru.lewis.point.api.RedisService;
+import ru.lewis.point.service.DatabaseServiceImpl;
+import ru.lewis.point.service.RedisServiceImpl;
 
 public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTask> implements ServerInfo, ChunkIOErrorReporter, CommandSource, ca.spottedleaf.moonrise.patches.chunk_system.server.ChunkSystemMinecraftServer { // Paper - rewrite chunk system
+    // lewis edit
+    private static final DatabaseService DATABASE_SERVICE = new DatabaseServiceImpl();
+    private static final RedisService REDIS_SERVICE = new RedisServiceImpl();
+
+    private static final int TEST = 0;
+
     private static MinecraftServer SERVER; // Paper
     public static final Logger LOGGER = LogUtils.getLogger();
     public static final net.kyori.adventure.text.logger.slf4j.ComponentLogger COMPONENT_LOGGER = net.kyori.adventure.text.logger.slf4j.ComponentLogger.logger(LOGGER.getName()); // Paper
@@ -748,8 +759,15 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         io.papermc.paper.plugin.lifecycle.event.LifecycleEventRunner.INSTANCE.callReloadableRegistrarEvent(io.papermc.paper.plugin.lifecycle.event.types.LifecycleEvents.COMMANDS, io.papermc.paper.command.brigadier.PaperCommands.INSTANCE, org.bukkit.plugin.Plugin.class, io.papermc.paper.plugin.lifecycle.event.registrar.ReloadableRegistrarEvent.Cause.INITIAL); // Paper - call commands event for regular plugins
         this.server.getCommandMap().registerServerAliases(); // Paper - relocate initial CommandMap#registerServerAliases() call
         ((org.bukkit.craftbukkit.help.SimpleHelpMap) this.server.getHelpMap()).initializeCommands();
-        this.server.getPluginManager().callEvent(new org.bukkit.event.server.ServerLoadEvent(org.bukkit.event.server.ServerLoadEvent.LoadType.STARTUP));
-        this.connection.acceptConnections();
+        // ru.lewis.point блять
+        CompletableFuture.allOf(
+            DATABASE_SERVICE.init(),
+            REDIS_SERVICE.init()
+        ).thenAccept(v -> {
+            PointAPI.init(DATABASE_SERVICE, REDIS_SERVICE);
+            this.server.getPluginManager().callEvent(new org.bukkit.event.server.ServerLoadEvent(org.bukkit.event.server.ServerLoadEvent.LoadType.STARTUP));
+            this.connection.acceptConnections();
+        });
     }
 
     public void initWorld(ServerLevel serverLevel, ServerLevelData serverLevelData, WorldData saveData, WorldOptions worldOptions) {
