From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zyr1xx <nadejdinkostya@gmail.com>
Date: Fri, 31 Jul 2026 01:26:41 +0300
Subject: [PATCH] upd: hook database/redis service


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 284947437eacf1d527b613d883c40bb3eb5faa90..f28cee515b5ccc023f790616b84fdbf953e617fe 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -1,6 +1,5 @@
 package net.minecraft.server;
 
-import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
 import com.mojang.datafixers.DataFixer;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Dynamic;
@@ -60,15 +59,23 @@ import net.minecraft.world.level.storage.LevelSummary;
 import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.WorldData;
 import org.slf4j.Logger;
+import ru.lewis.point.api.DatabaseService;
+import ru.lewis.point.api.PointAPI;
+import ru.lewis.point.api.RedisService;
+import ru.lewis.point.service.DatabaseServiceImpl;
+import ru.lewis.point.service.RedisServiceImpl;
 
 public class Main {
     private static final Logger LOGGER = LogUtils.getLogger();
+    public static final DatabaseService DATABASE_SERVICE = new DatabaseServiceImpl();
+    public static final RedisService REDIS_SERVICE = new RedisServiceImpl();
 
     @SuppressForbidden(
         reason = "System.out needed before bootstrap"
     )
     @DontObfuscate
     public static void main(final OptionSet optionSet) { // CraftBukkit - replaces main(String[] args)
+        PointAPI.init(DATABASE_SERVICE, REDIS_SERVICE);
         io.papermc.paper.util.LogManagerShutdownThread.hook(); // Paper - Improved watchdog support
         SharedConstants.tryDetectVersion();
         /* CraftBukkit start - Replace everything
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 047431262f0f6874a587b6db8032ee51f1015446..3477273329c3a1cb900aacfc1e18447ac2051448 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -150,7 +150,6 @@ import net.minecraft.world.level.levelgen.PhantomSpawner;
 import net.minecraft.world.level.levelgen.WorldOptions;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager;
 import net.minecraft.world.level.storage.CommandStorage;
-import net.minecraft.world.level.storage.DerivedLevelData;
 import net.minecraft.world.level.storage.DimensionDataStorage;
 import net.minecraft.world.level.storage.LevelData;
 import net.minecraft.world.level.storage.LevelResource;
@@ -161,17 +160,8 @@ import net.minecraft.world.level.storage.WorldData;
 import net.minecraft.world.phys.Vec2;
 import net.minecraft.world.phys.Vec3;
 import org.slf4j.Logger;
-import ru.lewis.point.api.DatabaseService;
-import ru.lewis.point.api.PointAPI;
-import ru.lewis.point.api.RedisService;
-import ru.lewis.point.service.DatabaseServiceImpl;
-import ru.lewis.point.service.RedisServiceImpl;
 
 public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTask> implements ServerInfo, ChunkIOErrorReporter, CommandSource, ca.spottedleaf.moonrise.patches.chunk_system.server.ChunkSystemMinecraftServer { // Paper - rewrite chunk system
-    // lewis edit
-    private static final DatabaseService DATABASE_SERVICE = new DatabaseServiceImpl();
-    private static final RedisService REDIS_SERVICE = new RedisServiceImpl();
-
     private static final int TEST = 0;
 
     private static MinecraftServer SERVER; // Paper
@@ -761,10 +751,9 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         ((org.bukkit.craftbukkit.help.SimpleHelpMap) this.server.getHelpMap()).initializeCommands();
         // ru.lewis.point блять
         CompletableFuture.allOf(
-            DATABASE_SERVICE.init(),
-            REDIS_SERVICE.init()
+            Main.DATABASE_SERVICE.init(),
+            Main.REDIS_SERVICE.init()
         ).thenAccept(v -> {
-            PointAPI.init(DATABASE_SERVICE, REDIS_SERVICE);
             this.server.getPluginManager().callEvent(new org.bukkit.event.server.ServerLoadEvent(org.bukkit.event.server.ServerLoadEvent.LoadType.STARTUP));
             this.connection.acceptConnections();
         });
